File: cs355/assign2/c/stats.c
#include "stats.h"


double dropRate(double oldRate, int oldDrops,
        double timeLastDrop, double newTime, 
        int newDrops){

    double newRate = (oldRate * oldDrops +
            (newTime - timeLastDrop)) / newDrops;

    return newRate;
}

double averageSysTime(double oldAvg, int lastCustCnt,
        double newTime, double startTime,
        int newCustCnt){

    double newAvg = ((oldAvg * lastCustCnt) +
            (newTime - startTime)) / newCustCnt;
    return newAvg;
}

double averageQueueLen(double oldAvg, double totalTime,
        int newLen, double startTime,
        double newTime){

    double newAvg = ((oldAvg*startTime) +
            (newLen * (newTime - startTime))) / totalTime;
    return newAvg;
}

File: cs355/assign2/python/stats.py

def dropRate(drops, time):
    return drops/(1.0 *time)

def avgSysTime(oldAvg, custCnt, startTime, newTime):
    return ((oldAvg * (custCnt-1)) + (newTime - startTime))/ (1.0*custCnt)

def avgQueueLen( oldAvg, newLen, startTime, newTime):
    return ((oldAvg * startTime) + (newLen * (newTime - startTime))) / (1.0*newTime)

def serverIdleTime ( oldIdleTime, startTime, endTime):
    return oldIdleTime + endTime - startTime



if __name__ == "__main__":
    print "testing dropRate 5 100"
    print dropRate(5, 100)


    print "testing avgSysTime 5, 100, 345, 350"
    print avgSysTime(5, 100, 345, 350)


    print "testing avgQueueLen(5, 7, 345, 350)"
    print avgQueueLen(5, 7, 345, 350)

    print "testing serverIdleTime( 89, 845, 850)"
    print serverIdleTime( 89, 845, 850)
